static/app/views/organizationGroupDetails/groupSimilarIssues/similarStackTrace/index.tsx

```diff
@@ -123,12 +64,56 @@ class SimilarStackTrace extends Component<Props, State> {
     }
 
     GroupingStore.onFetch(reqs);
-  }
+  }, [location.query, groupId, isUsingSimilarityViewV2, hasSimilarityFeature]);
+
+  const onGroupingChange = useCallback(
+    ({
+      mergedParent,
+      similarItems: updatedSimilarItems,
+      filteredSimilarItems: updatedFilteredSimilarItems,
+      similarLinks: updatedSimilarLinks,
+      loading,
+      error,
+    }) => {
+      if (updatedSimilarItems) {
+        setSimilarItems(updatedSimilarItems);
+        setFilteredSimilarItems(updatedFilteredSimilarItems);
+        setSimilarLinks(updatedSimilarLinks);
+        setLoading(loading ?? false);
+        setHasError(error ?? false);
+        return;
+      }
+
+      if (mergedParent && mergedParent !== groupId) {
+        // Merge success, since we can't specify target, we need to redirect to new parent
+        navigate(`/organizations/${orgId}/issues/${mergedParent}/similar/`);
+      }
+    },
+    [navigate, groupId, orgId]
+  );
+
+  useEffect(() => {
+    fetchData();
```

Comment:
If a component unmounts while a request is in flight, this will still trigger the group store callbacks. Is that ok or do we need to cleanup here?

Reply:
I think that's ok. It'll update the group store, but the component will not re-render because it will have stopped listening to the GroupingStore.

We're calling `GroupingStore.onFetch` which doesn't have any callbacks, so getting in there and stopping the update is going to be a bigger change.